src/context/HicetnuncContext.js

```diff
@@ -162,53 +165,39 @@ class HicetnuncContextProviderClass extends Component {
 
       },
 
-      // marketplace v2
-
-      collectv2: async (swap_id, xtz_amount) => {
-        return await Tezos.wallet
-          .at(this.state.v2)
-          .then((c) =>
-            c.methods
-              .collect(parseFloat(swap_id))
-              .send({
-                amount: parseFloat(xtz_amount),
-                mutez: true,
-                storageLimit: 310,
-              })
-          )
-      },
-
-      swapv2: async (from, royalties, xtz_per_objkt, objkt_id, creator, objkt_amount) => {
+      swapTeia: async (from, royalties, xtz_per_objkt, objkt_id, creator, objkt_amount) => {
         // If using proxy: both calls are made through this.state.proxyAddress:
         const objktsAddress = this.state.proxyAddress || this.state.objkts;
-        const marketplaceAddress = this.state.proxyAddress || this.state.v2;
+        const marketplaceAddress = this.state.proxyAddress || MARKETPLACE_CONTRACT_TEIA;
         const ownerAddress = this.state.proxyAddress || from;
 
         let objkts = await Tezos.wallet.at(objktsAddress)
         let marketplace = await Tezos.wallet.at(marketplaceAddress)
 
-        let list = [
+        const list = [
           {
             kind: OpKind.TRANSACTION,
-            ...objkts.methods.update_operators([{ add_operator: { operator: this.state.v2, token_id: parseFloat(objkt_id), owner: ownerAddress } }])
+            ...objkts.methods.update_operators([{ add_operator: { operator: MARKETPLACE_CONTRACT_TEIA, token_id: parseFloat(objkt_id), owner: ownerAddress } }])
               .toTransferParams({ amount: 0, mutez: true, storageLimit: 175 })
           },
           {
             kind: OpKind.TRANSACTION,
-            ...marketplace.methods.swap(creator, parseFloat(objkt_amount), parseFloat(objkt_id), parseFloat(royalties), parseFloat(xtz_per_objkt)).toTransferParams({ amount: 0, mutez: true, storageLimit: 300 })
+            ...marketplace.methods.swap(objktsAddress, parseFloat(objkt_id), parseFloat(objkt_amount), parseFloat(xtz_per_objkt), parseFloat(royalties), creator).toTransferParams({ amount: 0, mutez: true, storageLimit: 300 })
```

Comment:
In priciple they should be parseInt, but i understand to keep it like that

Reply:
yeah, there is a bunch of stuff that i know needs refactoring. but i didn't want to open the gates to hell as part of this pr :-)